internal/policy/root.go

```diff
@@ -27,35 +29,49 @@ func InitializeRootMetadata(key *tuf.Key) *tuf.RootMetadata {
 	return rootMetadata
 }
 
-// AddTargetsKey adds targetsKey as a trusted public key in rootMetadata for the
+// AddTargetsKey function is used to add the 'targetsKey' as a trusted public key in 'rootMetadata' for the
 // top level Targets role.
-func AddTargetsKey(rootMetadata *tuf.RootMetadata, targetsKey *tuf.Key) *tuf.RootMetadata {
+func AddTargetsKey(rootMetadata *tuf.RootMetadata, targetsKey *tuf.Key) (*tuf.RootMetadata, error) {
+	if rootMetadata == nil {
+		return nil, fmt.Errorf("rootMetadata is nil")
+	}
+	if targetsKey == nil {
+		return nil, fmt.Errorf("targetsKey is nil")
```

Comment:
The same thing here, the targetsKey instance is explicitly loaded by the repository API, which aborts prior to this if the key isn't loaded correctly...

Reply:
same here re named errors